scripts/Synapse/analyzeLinkageAlgorithms.ipynb

```diff
@@ -189,6 +257,46 @@
         "    print(\"Already mounted\")"
       ]
     },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Extracting the MPI\n",
+        "Now we get into the meat of the notebook, in which we read and analyze data from the MPI. The following cells open a connection to the MPI specified by the configuration options above, extract patient data from it, and format it into an appropriately sized sample for use with the remainder of this notebook. All 3 notebook modes rely on clean, properly formatted patient data pulled from the MPI. This is accomplished with `pyspark` using access patterns called \"temporary database views\". We can read a dataset distributed across multiple tables into memory in parallel, and by constructing specific views for each of the tables we want to access, `pyspark` provides us a window to run queries or perform joins against that table without disrupting the underlying information actually stored in the MPI. Once we have these views, we can begin with the default `patient` table and iteratively join each other table to the assorted patient records there, ensuring that their information is properly grouped and collated along the way, so that our result is a single massaged table consisting of one patient row per re-constructed health care record."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "# MPI ACCESS\n",
+        "# Create views into all tables in the MPI so we can extract them in parallel\n",
+        "# Each of these views represents a safe \"copy\" of one table in the MPI that we\n",
+        "# can query and/or manipulate without worrying about affecting the actual data\n",
+        "# stored in the database. We need one view for each table we want to access.\n",
+        "patient_view = \"patient_view\"\n",
+        "mpi_patient_data = spark.read.jdbc(url, DB_TABLE_PATIENT, properties=db_props)\n",
```

Comment:
Oh 100%. Fighting the SQL through spark is a losing game, especially because it doesn't have access to all of the SQL efficiency or even options you'd want. If you can get it in a DF, though, you're off to the races.